### ISSUE TITLE
`SignSGD` accepts `weight_decay` but never applies it

### ISSUE THREAD

**Issue by 11NOel11** [kozistr/pytorch_optimizer#517]
## Describe the bug

`SignSGD` accepts `weight_decay` and `weight_decouple`, validates them, and stores them in
`defaults`, but neither value is used anywhere in the step path.

- `sgd.py:429` validates `weight_decay`, and `sgd.py:437-438` store both in `defaults`.
- `_step_foreach` (`sgd.py:470`) and `_step_per_param` (`sgd.py:487`) never call
  `apply_weight_decay_foreach` or `apply_weight_decay`.
- The base helper at `base/optimizer.py:135` is not applied automatically. `SGDW` calls it
  explicitly at `sgd.py:205` and `sgd.py:242`.

So `SignSGD(params, lr=..., weight_decay=0.01)` runs without any weight decay and gives no warning.

I am filing this separately from #516 because fixing it changes results for anyone currently
passing a non-zero `weight_decay`, so it seemed better to ask than to send a patch. If you want the
fix, `Tiger` looks like the right template rather than `SGDW`: it is also a sign-based update, and
it applies decay after gradient maximisation and before the moment update (`tiger.py:92-98` and
`tiger.py:116-126`). Applying it after the buffer update instead leaves coupled decay outside the
`sign()` and it has no effect. I have a working patch and tests if that is useful.

## To Reproduce

* OS : macOS
* PyTorch version : 2.9.1
* Python version : 3.12
* pytorch-optimizer version : 3.10.1 (also reproduced on `main` at b452a0f)
* reproducible codes :

```python
import torch
from pytorch_optimizer import SignSGD

for weight_decouple in (True, False):
    for foreach in (False, True):
        p = torch.nn.Parameter(torch.full((2,), 2.0))
        opt = SignSGD([p], lr=0.1, momentum=0.9, weight_decay=0.5,
                      weight_decouple=weight_decouple, foreach=foreach)
        p.grad = torch.zeros(2)
        opt.step()
        print(f"weight_decouple={weight_decouple} foreach={foreach} -> p={p[0].item():.4f}")
```

Output:

```
weight_decouple=True foreach=False -> p=2.0000
weight_decouple=True foreach=True  -> p=2.0000
weight_decouple=False foreach=False -> p=2.0000
weight_decouple=False foreach=True  -> p=2.0000
```

With decoupled decay at `lr=0.1` and `weight_decay=0.5`, the parameter should move from `2.0` to
`1.9`. It does not move at all.

**Comment by kozistr** [comment]
@11NOel11 hi! thanks for the report. yes, as you mentioned, weight decay should be applied to SignSGD.

if you have the bandwidth, could you make a patch for this?

thanks!
